src/types.ts

```diff
@@ -117,6 +117,14 @@ export type PerFileSignOptions = {
  * @internal
  */
 export type OnlySignOptions = {
+  /**
+   * Combine all files with identical CLI arguments in a single `codesign` call instead of running
+   * `codesign` once for each of those files. This minimizes the number of child processes that need
+   * to be spawned and reduces the time it takes to sign all files.
+   *
+   * @defaultValue `false`
+   */
+  batchCodesignCalls?: boolean;
```

Comment:
I don't think we define "experimental" APIs formally in the ecosystem, but I think it'd be good to mark this as experimental anyways to give us some leeway with behaviour.

```suggestion
  /**
   * Combine all files with identical CLI arguments in a single `codesign` call instead of running
   * `codesign` once for each of those files. This minimizes the number of child processes that need
   * to be spawned and reduces the time it takes to sign all files.
   *
   * @defaultValue `false`
   * @experimental
   */
  batchCodesignCalls?: boolean;
```